### ISSUE TITLE
Building null origins in RelyingParty

### ISSUE THREAD

**Issue by philsmart** [Yubico/java-webauthn-server#347]
I think you should be able to build the [RelyingParty](https://github.com/Yubico/java-webauthn-server/blob/main/webauthn-server-core/src/main/java/com/yubico/webauthn/RelyingParty.java) object using a null Origins Set (it just takes the RP ID if it is). However, the builder will not allow it `Caused by: java.lang.NullPointerException: origins is marked non-null but is null`. I guess that is because of the nonnull annotation on the origins field.

Version 2.5.0.

**Comment by emlun** [comment]
Sorry, I don't understand what you mean by this:

> (it just takes the RP ID if it is)

What are you trying to accomplish?

**Comment by philsmart** [comment]
If I want to build the RelyingParty object e.g. 
```
final RelyingParty rp = RelyingParty.builder().identity(
                    RelyingPartyIdentity
                    .builder()
                    .id(getRelyingPartyId())
                    .name(getRelyingPartyName())
                    .build()).credentialRepository(getCredentialRepository())
            .allowOriginPort(isAllowOriginPort())
            .allowOriginSubdomain(isAllowOriginSubdomain())
            .origins(getOrigins())
            .build();
```
If getOrigins is null, the builder fails with an NPE. I want to pass null through the builder, the constructor seems to allow null origins, and if it is null it sets the identity.getId()
```
    this.origins =
        origins != null
            ? CollectionUtil.immutableSet(origins)
            : Collections.singleton("https://" + identity.getId());
```
Otherwise, I have to build the Relying Party object conditionally.

**Comment by philsmart** [comment]
Thanks for the quick reply. Yes, I am currently handling it conditionally, e.g. if the origins are not set (configured on my service) I just leave it out, and if it is set I call it. It would be easier not to make that decision when I am building the RelyingParty object, and just send through a null and let the builder handle it. 

But I get it is a convenience thing, so feel free to close this issue if it is not something you'd want to change.

**Comment by emlun** [comment]
Fair enough, I had the same thought while looking into this. We'll consider it for the next release!

**Comment by philsmart** [comment]
Thank you.

**Comment by emlun** [comment]
Would an `origins(Optional<Set<String>>)` setter overload suffice for your use case?

I'm not quite convinced of simply dropping the `@NonNull` requirement, because this parameter in particular is very likely to be read from an environment variable or similar, and it's fairly likely that whatever function reads that may return `null` if the environment variable isn't set, for example. A silent fallback to the default `origins` in that case may be incorrect behaviour that's unnecessarily hard to diagnose. So I'd therefore prefer an `Optional` setter, since passing `Optional.empty` (for example by simply wrapping with `Optional.ofNullable`) is a better indication than `null` of the intent to fall back to the default value.

**Comment by philsmart** [comment]
Absolutely, that makes sense and works for me. Thanks!
